processor.go

```diff
@@ -27,7 +27,7 @@ import (
 type Processor interface {
 	// Specification contains the metadata of this processor like name, version,
 	// description and a list of parameters expected in the configuration.
-	Specification() Specification
+	Specification() (Specification, error)
```

Comment:
When would this function return an error? 🤔

Reply:
That's for cases when Conduit cannot communicate with a standalone processor for whatever reason (processor not running, error in serialization, etc.)

Reply:
Hmm, that sounds like we're adjusting the interface because of internal communication between Conduit and the processor 😕 I mean, I understand why it's here, doesn't mean I like it 😆 If we could hide this error internally it would make more sense from the perspective of the processor implementation.